connector.go

```diff
@@ -2,49 +2,68 @@ package tarantool
 
 import (
 	"net/url"
-	"sync"
+	"context"
 )
 
 type Connector struct {
-	sync.Mutex
 	RemoteAddr string
 	options    Options
 	conn       *Connection
+	ch         chan struct{}
 }
 
 // New Connector instance.
 func New(dsnString string, options *Options) *Connector {
+	ch := make(chan struct{}, 1)
+	ch <- struct{}{}
+
 	if options != nil {
-		return &Connector{RemoteAddr: dsnString, options: *options}
+		return &Connector{RemoteAddr: dsnString, options: *options, ch: ch}
 	}
-	return &Connector{RemoteAddr: dsnString}
+	return &Connector{RemoteAddr: dsnString, ch: ch}
 }
 
 // Connect returns existing connection or will establish another one.
 func (c *Connector) Connect() (conn *Connection, err error) {
-	c.Lock()
+	return c.ConnectContext(context.Background())
+}
+
+// Connect returns existing connection or will establish another one.
+func (c *Connector) ConnectContext(ctx context.Context) (conn *Connection, err error) {
+	doneChan := ctx.Done()
```

Comment:
It could be simplified to:
```
  select {
  case <-ctx.Done():
   return nil, ctx.Err()
  case <-c.ch:
  }
```
It is not necessary to check if channel is nil because nil channel just "turn off" the case.